packages/terra-collapse/src/Collapse.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Arrange from 'terra-arrange';
+import Button from 'terra-button';
+import IconChevronRight from 'terra-icon/lib/icon/IconChevronRight';
+import Toggler from 'terra-toggler';
+import 'terra-base/lib/baseStyles';
+import './Collapse.scss';
+
+const propTypes = {
+  /**
+   * Content in the body of the collapse component that will be expanded or collapsed
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Sets the text inside of the button when the collapse component is closed
+   */
+  closedButtonText: PropTypes.node.isRequired,
+  /**
+   * Icon displayed next to text content within the collapse button
+   */
+  icon: PropTypes.element.isRequired,
+  /**
+   * Used to set props and HTML attributes on the collapse button
+   */
+  buttonAttrs: PropTypes.object,
+  /**
+   * Sets the collapse to be animated when it is opened or closed
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the collapse initial state to open
+   */
+  isInitiallyOpen: PropTypes.bool,
+  /**
+   * Callback function triggered when collapse is opened
+   */
+  onOpen: PropTypes.func,
+  /**
+   * Callback function triggered when collapse is closed
+   */
+  onClose: PropTypes.func,
+  /**
+   * Sets the text inside of the button when the collapse component is open
+   */
+  openedButtonText: PropTypes.node,
+};
+
+const defaultProps = {
+  isAnimated: false,
+  isInitiallyOpen: false,
+  icon: <IconChevronRight />,
+};
+
+class Collapse extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: this.props.isInitiallyOpen,
+    };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
I think we want to compare `nextProps.isOpen` with `this.props.isOpen` to ensure props did actually change. Then only setState when props have actually changed.

https://facebook.github.io/react/blog/2016/01/08/A-implies-B-does-not-imply-B-implies-A.html